Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient.m

```diff
@@ -57,6 +55,8 @@
 #define kMinSuccessfulStatusCode 200
 #define kMaxSuccessfulStatusCode 299
 
+#define BUYAssertCheckout(checkout) NSAssert([(checkout) hasToken], @"Checkout assertion failed. Checkout must have a valid token associated with it.")
```

Comment:
We should throw an explicit exception, because `NSAssert()` is disabled in release builds, but we should always complain/crash if we are passed an invalid checkout.